### PR TITLE
[Spring]feat: Add Outbound Type

### PR DESCRIPTION
**Pull request by YinglueZhang-MS**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
spring

File: src/spring/azext_spring/_params.py [hunk 1]
```diff
@@ -247,6 +251,25 @@ def load_arguments(self, _):
...
+                   arg_type=get_enum_type(SessionAffinity),
+                   help='Ingress session afiinity of app.',
+                   validator=validate_ingress_timeout)
+        c.argument('session_max_age',
+                   type=int,
+                   help='Time until the cookie expires. Minimum is 1 second, maximum is 7 days. If set to 0, the expiration period is equal to the browser session period.',
+                   validator=validate_ingress_session_max_age)
+        c.argument('backend_protocol',
+                   arg_type=get_enum_type(BackendProtocol),
+                   help='Ingress backend protocol of app.')
```
Comment (zhoxing-ms) [inline src/spring/azext_spring/_params.py:272]: Can we use the `for` loop statement for create and update commands to reuse these same parameter definitions?

Comment (zhoxing-ms) [inline src/spring/azext_spring/_params.py:272]: Could we use the `for` loop statement to avoid repeatedly defining the same parameters

**Comment by zhoxing-ms** [comment]
Please write the description of PR change into the `HISTORY.rst`. In addition, if you want to release a new extension version for this PR, please also upgrade the `VERSION` in `setup.py`

File: src/spring/azext_spring/_params.py (hunk 1 above)
Comment (CaihuaRui) [reply src/spring/azext_spring/_params.py]: Updated, thanks.

File: src/spring/HISTORY.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 Release History
 ===============
+1.2.0
```
Comment (zhoxing-ms) [inline src/spring/HISTORY.md:3]: May I ask why you upgraded the Minor version (1.2.0) instead of Patch version (1.1.6) ?

Comment (CaihuaRui) [reply src/spring/HISTORY.md]: Updated to 1.1.6.

File: src/spring/HISTORY.md [hunk 3]
```diff
@@ -2,6 +2,8 @@ Release History
 ===============
 1.1.6
 ---
+* Command `az spring create` and ` has new argument "--outboundType" to enable UDR for VNet instance.
+* Command `az spring app create` and `az spring app update` has new argument "--ingress-read-timeout", "--ingress_send_timeout", "--session_affinity", "--session_max_age", "--backend_protocol" to customize the ingress settings of user applications.
```
Comment (zhoxing-ms) [inline src/spring/HISTORY.md:6]:
```suggestion
* Command `az spring app create` and `az spring app update` has new argument "--ingress-read-timeout", "--ingress_send_timeout", "--session_affinity", "--session-max-age", "--backend-protocol" to customize the ingress settings of user applications.
```

**Comment by CaihuaRui** [comment]
Don't modify setup.py for now, we still need to wait for a parameter about ingress from other colleagues

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**